### DISCUSSION TITLE
I want to understand how the source code is compiled from text.

### DISCUSSION THREAD

**Discussion by Abijeet123** [carbon-language/carbon-lang#2644] [Explorer]
I wanted to build a REPL for the language. I would like to know how the source code is compiled from the text. You can just point out what are the functions involved for lexical analysis to parsing it to building the ast and finally executing the ast. Thanks and regards.

**Comment by geoffromer** [comment]
The [README](https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/README.md) is probably the best place to go for that kind of overview. Some of the subdirectories of `explorer/` also have their own README files. We use Flex and Bison for lexical analysis and parsing, so for those phases you might want to look at the documentation for those tools if you're not already familiar with them.
